src/app/vertextool/qgsvertextool.cpp

```diff
@@ -984,7 +984,8 @@ QgsGeometry QgsVertexTool::cachedGeometry( const QgsVectorLayer *layer, QgsFeatu
   {
     connect( layer, &QgsVectorLayer::geometryChanged, this, &QgsVertexTool::onCachedGeometryChanged );
     connect( layer, &QgsVectorLayer::featureDeleted, this, &QgsVertexTool::onCachedGeometryDeleted );
-    // TODO: also clear cache when layer is deleted
+    connect( layer, &QgsVectorLayer::willBeDeleted, this, &QgsVertexTool::clearGeometryCache );
+    connect( layer, &QgsVectorLayer::dataChanged, this, &QgsVertexTool::clearGeometryCache );
```

Comment:
Is this signal emitted when features are added/deleted/changed? Because we don't want to force a full invalidation on these operations

Reply:
AFAIK, no, dataChanged is only emitted on a change of subset string on a layer or if features are added/deleted/changed on another layer that has been declared to be a dependency

Reply:
Thanks -- I've updated the dox in 9df3eb3e5cae9ce1df4dd3e0554989c38f9ead2c to explicitly state this!

Reply:
Thanks for the doc clarification !